aspnetcore/razor-components/components.md

```diff
@@ -761,3 +761,54 @@ The time is 10/04/2018 01:26:52.
 
 Your pet's name is Rex.
 ```
+
+## Dynamic components
+
+Dynamic components can be used to dynamically generate content. `Microsoft.AspNetCore.Components.RenderTree` provides methods for manipulating components and elements.
+
+> [!WARNING]
+> Dynamic components are an advanced scenario and aren't recommended for general use. Dynamically generated components can bypass tag structure safety checks. Inadvertently rendering a faulty tag in the browser can result in a security vulnerability.
```

Comment:
What is the security vulnerability and safety check being referred to here?

Reply:
Discussed with Steve while there: He suggested this exact phrasing ("an advanced scenario" ... from my notes of the meeting). I inquired about a malformed tag escaping into the rendered output that might lead to a security vulnerability, and he said it's possible for that to happen.

Reply:
I wouldn’t refer to a security issue unless there is a specific known one. I guess I was unclear, but my intent was to say you can just cause undefined behaviour if your tags don’t match.

Reply:
Change to a NOTE then? ... and soften the lingo?

Reply:
Up to you - I just suggest not conflating this with security at all, as that would be misleading.

Reply:
I agree that dynamic components is an advanced scenario. But the security warning doesn't provide enough information to know that I am avoiding the unspecified vulnerability. What does it mean to render a "faulty tag" and what is the resulting security vulnerability? How do I know that I haven't rendered a faulty tag? How do I know if I've mitigated the vulnerability? What "safety check" do I need to replicate?